From 35fc34271fbbd6f504aa428e7da45bfb3c8b7a44 Mon Sep 17 00:00:00 2001
From: Anatol Pomozov <anatol.pomozov@gmail.com>
Date: Mon, 29 Jun 2015 19:11:19 -0700
Subject: [PATCH 3/4] Fix compile error
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

/home/anatol/sources/dreamstudio/DSView/DSView/pv/mainwindow.cpp: In member function ‘bool pv::MainWindow
/home/anatol/sources/dreamstudio/DSView/DSView/pv/mainwindow.cpp:742:77: error: conversion from ‘uint64_t
             s_obj["vdiv"] = QJsonValue::fromVariant(dsoSig->get_vDialValue());
                                                                             ^
---
 DSView/pv/mainwindow.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/DSView/pv/mainwindow.cpp b/DSView/pv/mainwindow.cpp
index a00cc67..00ef817 100644
--- a/DSView/pv/mainwindow.cpp
+++ b/DSView/pv/mainwindow.cpp
@@ -739,8 +739,8 @@ bool MainWindow::store_session(QString name)
         s_obj["strigger"] = s->get_trig();
         boost::shared_ptr<view::DsoSignal> dsoSig;
         if (dsoSig = dynamic_pointer_cast<view::DsoSignal>(s)) {
-            s_obj["vdiv"] = QJsonValue::fromVariant(dsoSig->get_vDialValue());
-            s_obj["vfactor"] = QJsonValue::fromVariant(dsoSig->get_factor());
+            s_obj["vdiv"] = QJsonValue::fromVariant(static_cast<qulonglong>(dsoSig->get_vDialValue()));
+            s_obj["vfactor"] = QJsonValue::fromVariant(static_cast<qulonglong>(dsoSig->get_factor()));
             s_obj["coupling"] = dsoSig->get_acCoupling();
             s_obj["trigValue"] = dsoSig->get_trigRate();
             s_obj["zeroPos"] = dsoSig->get_zeroRate();
-- 
2.4.4

